src/electron/office/document_client.cc

```diff
@@ -310,6 +324,71 @@ void DocumentClient::HandleStateChange(std::string payload) {
   }
 }
 
+void DocumentClient::OnClipboardChanged() {
+  std::vector<std::string> mime_types;
+
+  mime_types.push_back("text/plain;charset=utf-8");
+  mime_types.push_back("image/png");
+
+  std::vector<const char*> mime_c_str;
+
+  for (const std::string& mime_type : mime_types) {
+    mime_c_str.push_back(mime_type.c_str());
+  }
+  mime_c_str.push_back(nullptr);
+
+  size_t out_count;
+  char** out_mime_types = nullptr;
+  size_t* out_sizes = nullptr;
+  char** out_streams = nullptr;
+
+  bool success = document_->getClipboard(
+      mime_types.size() ? mime_c_str.data() : nullptr, &out_count,
+      &out_mime_types, &out_sizes, &out_streams);
+
+  if (!success) {
+    LOG(ERROR) << "Unable to get document clipboard";
+    return;
+  }
+
+  ui::ScopedClipboardWriter writer(ui::ClipboardBuffer::kCopyPaste);
+
+  for (size_t i = 0; i < mime_types.size(); ++i) {
+    size_t buffer_size = out_sizes[i];
+    std::string mime_type = out_mime_types[i];
+    if (buffer_size > 0) {
```

Comment:
Better to `if(buffer_sizer <= 0) break;` here. Prefer to return early for pre-condition checks

Reply:
I think it needs to be a `continue` instead of `break` since there could be other mimetypes that have a size but I agree that is a better approach